src/interrupts/mag_int.rs

```diff
@@ -0,0 +1,409 @@
+            FLAG::Enabled => data | 1,       // if Enabled, set bit
+            FLAG::Disabled => data,                 // if Disabled, bit is cleared
+        };
+
+        self.interface.write(Sensor::Magnetometer, register::Mag::INT_CFG_M.addr(), data)?;
+
+        Ok(())
+
+    }
+
+
+    /// Get all the flags from the INT_SRC_M register
+    pub fn mag_int_status(&mut self) -> Result<IntStatusMag, T::Error> {        
+        
+        let reg_data: u8 = self.read_register(Sensor::Magnetometer, register::Mag::INT_SRC_M.addr())?;
+
+        let status = IntStatusMag {            
+            /// Does value on X-axis exceed the threshold on the positive side?
+            xaxis_exceeds_thresh_pos: match reg_data & M_INT_Bitmasks::PTH_X {
+                0 => false,
+                _ => true,
+            },
+            /// Does value on Y-axis exceed the threshold on the positive side?
+            yaxis_exceeds_thresh_pos: match reg_data & M_INT_Bitmasks::PTH_Y {
+                0 => false,
+                _ => true,
+            },
+            /// Does value on Z-axis exceed the threshold on the positive side?
+            zaxis_exceeds_thresh_pos: match reg_data & M_INT_Bitmasks::PTH_Z {
+                0 => false,
+                _ => true,
+            },
+            /// Does value on X-axis exceed the threshold on the negative side?
+            xaxis_exceeds_thresh_neg: match reg_data & M_INT_Bitmasks::NTH_X {
+                0 => false,
+                _ => true,
+            },
+            /// Does value on Y-axis exceed the threshold on the negative side?
+            yaxis_exceeds_thresh_neg: match reg_data & M_INT_Bitmasks::NTH_Y {
+                0 => false,
+                _ => true,
+            },
+            /// Does value on Z-axis exceed the threshold on the negative side?
+            zaxis_exceeds_thresh_neg: match reg_data & M_INT_Bitmasks::NTH_Z {
+                0 => false,
+                _ => true,
+            },
+            /// Did internal measurement range overflow on magnetic value?
+            measurement_range_overflow: match reg_data & M_INT_Bitmasks::MROI {
+                0 => false,
+                _ => true,
+            },
+            /// This bit signals when the interrupt event occurs.
+            interrupt_occurs: match reg_data & M_INT_Bitmasks::INT {
+                0 => false,
+                _ => true,
+            },
+        };
+        Ok(status)
+    }
+
+
+    // == MAKE SURE THE ORDER IS CORRECT!!! == 
+
+    /// Set threshold in miligauss
+    pub fn set_mag_threshold(&mut self, threshold: f32) -> Result<(), T::Error> {
+        let sensitivity = self.mag.scale.sensitivity();
+        let mut data = threshold / sensitivity;
+        // make sure it's not more than 15 bits, and it must be a positive value
+        if data >= 32767.0 {
+            data = 32767.0;
+        } else if data < 0.0 {
+            data = 0.0;
+        }
+        //data = data as u16;
```

Comment:
This line doesn't do anything. We should remove.

Reply:
You mean the commented one, right?